### DISCUSSION TITLE
Check if an array contains every value of another array.

### DISCUSSION THREAD

**Discussion by caseywalker** [nss-evening-cohort-14/evening-client-side#207] [Look at what I learned!]
# Context
I am currently working through the Nutshell project. Our app is for a restaurant, and we have our Menu Items listed out. We are adding a feature to allow our users to filter through our menu items based on available Ingredients. 

# The data required to make it work
We have a dropdown of all available ingredients, that is being stored in an array. We are then taking that array and passing it to a call to our database to filter through menu items based on those ingredients. 
* Array 1 = Ingredients selected from the dropdown
* Array 2 = All available menu items. 

# The challenge
* How do we compare the array of ingredients we pass to our DB, and only return menu items that contain EVERY ingredient selected?

# The solution in a Nutshell (pun intended)
* First, the .filter on all menu items, and THEN the .every operator. 
* What is the .every operator? It loops through every element on the array it is called on, and returns true if every element within array 1 meets a certain logic (here, we are checking if array 2 (Menu Items) includes every element of array 1 (Ingredients). 

# Code from our call to firebase:
```
const getFilterMenuItems = (ingredientArray) => new Promise((resolve, reject) => {
  getMenuItems().then((menuArray) => {
    const result = menuArray.filter((item) => ingredientArray.every((ingredient) => item.ingredients.includes(ingredient)));
    resolve(result);
  }).catch((error) => reject(error));
}); 
```

Special thanks to @TrinityChristiana for helping me think through this logic and pointing me in the right direction.

**Comment by drteresavasquez** [comment]
@caseywalker THIS IS FANTASTIC!!! 🥳

**Reply by danicrosby** [reply]
Nice explained, Casey! Appreciate this post.
